### PR TITLE
Fix S1871: "Branches with same implementation" should ignore single l…

### PR DESCRIPTION
**Pull request by valhristov**
…ine blocks

### CODE REVIEW

File: sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ConditionalStructureSameImplementation.cs [hunk 1]
```diff
@@ -67,42 +73,50 @@ protected sealed override void Initialize(SonarAnalysisContext context)
             context.RegisterSyntaxNodeActionInNonGenerated(
                 c =>
                 {
-                    var switchSection = (SwitchSectionSyntax) c.Node;
+                    var switchSection = (SwitchSectionSyntax)c.Node;
+
+                    if (GetStatements(switchSection).Where(IsNotIgnored).AreLessThan(2))
```
Comment (Evangelink) [inline sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ConditionalStructureSameImplementation.cs]: Might be worth creating a meaningful variable before the `if` statement.

File: sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ConditionalStructureSameImplementation.cs [hunk 2]
```diff
@@ -67,42 +73,50 @@ protected sealed override void Initialize(SonarAnalysisContext context)
...
+                        ReportSyntaxNode(c, switchSection, precedingSection, "case");
                     }
                 },
                 SyntaxKind.SwitchSection);
         }
 
-        private static void CheckStatement(SyntaxNodeAnalysisContext context, StatementSyntax statementToCheck,
+        private static IEnumerable<StatementSyntax> GetStatements(SwitchSectionSyntax switchSection)
+        {
+            return Enumerable.Empty<StatementSyntax>()
```
Comment (Evangelink) [inline sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ConditionalStructureSameImplementation.cs]: Do you really need the first `Enumerable.Empty<>().Union()`? Can't you start directly with the content?

File: sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ConditionalStructureSameImplementation.cs [hunk 3]
```diff
@@ -67,42 +73,50 @@ protected sealed override void Initialize(SonarAnalysisContext context)
...
+                .Union(switchSection.Statements.OfType<BlockSyntax>().SelectMany(block => block.Statements))
+                .Union(switchSection.Statements.Where(s => !s.IsKind(SyntaxKind.Block)));
+        }
+
+        private static void CheckStatement(SyntaxNodeAnalysisContext context, StatementSyntax statement,
             IEnumerable<StatementSyntax> precedingStatements)
         {
+            if (!statement.ChildNodes().AreMoreThan(1))
```
Comment (Evangelink) [inline sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ConditionalStructureSameImplementation.cs]: I would negate the condition and would probably extract it into some variable before the `if` condition.

File: sonaranalyzer-dotnet/src/SonarAnalyzer.VisualBasic/Rules/ConditionalStructureSameImplementation.cs [hunk 4]
```diff
@@ -116,5 +128,10 @@ protected override void Initialize(SonarAnalysisContext context)
             context.ReportDiagnostic(Diagnostic.Create(rule,
                 location, locationProvider.First().GetLineNumberToReport(), constructType));
         }
+
+        private static bool IsNotIgnored(StatementSyntax statement)
```
Comment (michalb-sonar) [inline sonaranalyzer-dotnet/src/SonarAnalyzer.VisualBasic/Rules/ConditionalStructureSameImplementation.cs]: Can this be positive? For example rename to IsIgnored or IsIgnoredStatement and then negate at the point of calling.

**Pull request merged**